feat: Add validation to timestamp and tag fields

sravan-s opened the pull request:
Timestamp and tag fields shouldnt start or end with dot
Todo: Implement tests

Fixes: https://github.com/quickwit-oss/quickwit/issues/3538

sravan-s committed:
feat: Add validation to timestamp and tag fields

Timestamp and tag fields shouldnt start or end with dot
Todo: Implement tests

AyWa commented:
Also as I understand we should escape the field if it contains dot ? from the issue ` And if a dot is part of a field name, it must be escaped.`

sravan-s commented:
That they did already, AFAIK ~ https://github.com/quickwit-oss/quickwit/pull/3537/files#diff-dc818f14abb5998cbc18bdb4ba5985f7bf6d3c0fa8c765e80c2149cd8a336f7bR577-R587

AyWa commented:
@sravan-s yeah thats right !

AyWa committed:
ensure we validate escaped string

AyWa committed:
add test for tag validation

AyWa committed:
fix linter

sravan-s committed:
Endwith dont need to be checked with string literal

`//.` and `.` should end in the same way

Also address review comment about `['//.']` -> `"//."`

fulmicoton committed:
Added support for exists query, as defined in Elasticsearch (#3710)

Field exists does not consider types, only field names.
Field capability will have to be handled differently unfortunately.

This works by introducing an internal (but normal) "u64" field
that stores postings list for field existence.

For performance/RAM reasons, the fields full path is not stored
as a string but instead we compute a u64-fnv hash using the
path from root to leaf.

If the hash perfects ideally, even with the anniversary attach, collisions
are very unlikely.

When dealing with complex JSON with the raw tokenizer this feature can
double the number of tokens we deal with, and has an impact on
performance.

For this reason, it is not added as an option in the DocMapper.

Like Elasticsearch, we only store field existence of indexed fields.
Also in order to handle refinement like expand_dots,
we work over the built tantivy Document and reuse the existing
resolution logic.

On 1.4GB of gharchive (which is close to a worst case scenaio),
see the following performance/index size change:

With field_exists enabled
- Indexing Throughput: 41 MB/s
- Index size: 701M

With field_exists disabled
- Indexing Throughput: 46 MB/s
- Index size: 698M

fulmicoton committed:
Merge branch 'main' into feat/validate-tag-timestamp
